erc721-tests/contracts/Erc-721.sol

```diff
@@ -44,24 +43,26 @@ contract Erc721 is Ownable, ERC721("SolicyNFT", "HNFT") {
         return _symbol;
     }
 
-    function _baseURI() internal view virtual override returns (string memory) {
+    function baseURI() public view virtual returns (string memory) {
         return __baseURI;
     }
 
     function tokenURI(uint256 tokenId_) public view virtual override returns (string memory) {
         _requireMinted(tokenId_);
 
-        return bytes(_baseURI()).length > 0 ? string(abi.encodePacked(_baseURI(), Strings.toString(tokenId_), uriSuffix)) : "";
+        return bytes(baseURI()).length > 0 ? string(abi.encodePacked(baseURI(), Strings.toString(tokenId_), ".json")) : "";
     }
 
     function mintToken(address recipient, string memory name_) onlyOwner public {
         require(owner()!=recipient, "Recipient cannot be the owner of the contract");
         _safeMint(recipient, tokenId);
-        ownershipRecord[recipient].push(TokenMetaData(name_, tokenId, 0, 0, tokenURI(tokenId), 0, 0));
+
+        metadataRecord[tokenId] = TokenMetaData(name_, tokenId, 0, 0, tokenURI(tokenId), 0, 0);
+
         tokenId = tokenId + 1;
```

Comment:
Try to use totalSupply not [counter variable](https://stackoverflow.com/questions/71708142/gas-efficiency-of-totalsupply-vs-a-tokenid-counter-erc-721)